tests/robot-tests/tests/public_api/public_api_restricted.robot

```diff
@@ -0,0 +1,300 @@
+    user edits release status
+    user checks checklist errors contains
+    ...    1 issue that must be resolved before this release can be published.
+    user checks checklist errors contains link
+    ...    All public API data set mappings must be completed
+
+Create a third draft release via api
+    user navigates to publication page from dashboard    ${PUBLICATION_NAME}
+    user creates release from publication page    ${PUBLICATION_NAME}    Academic year    3020
+
+Upload subject to third release
+    user uploads subject    ${SUBJECT_NAME_5}    institution_and_provider.csv    institution_and_provider.meta.csv
+
+Add data guidance to third release
+    user clicks link    Data and files
+    user waits until h2 is visible    Add data file to release
+
+    user clicks link    Data guidance
+    user waits until h2 is visible    Public data guidance
+
+    user waits until page contains element    id:dataGuidance-dataFiles
+    user waits until page contains accordion section    ${SUBJECT_NAME_5}
+
+    user enters text into data guidance data file content editor    ${SUBJECT_NAME_5}
+    ...    ${SUBJECT_NAME_5} Main guidance content
+
+
+Save data guidance
+    user clicks button    Save guidance
+
+Create a different version of API dataset (major version)
+    user scrolls to the top of the page
+    user clicks link    API data sets
+    user waits until h2 is visible    API data sets
+
+    user waits until h3 is visible    Current live API data sets
+
+    user checks table column heading contains    1    1    Version    xpath://table[@data-testid="live-api-data-sets"]
+    user clicks button in table cell    1    3    Create new version    xpath://table[@data-testid="live-api-data-sets"]
+
+    ${modal}=    user waits until modal is visible    Create a new API data set version
+
+    user chooses select option    id:apiDataSetCreateForm-releaseFileId   ${SUBJECT_NAME_5}
+    user clicks button    Confirm new data set version
+
+
+    user waits until page finishes loading
+    user waits until modal is not visible    Create a new API data set version    %{WAIT_LONG}
+
+Validate the summary contents inside the 'draft version details' table
+    user waits until h3 is visible    Draft version details
+    user waits until element contains    css:dl[data-testid="draft-version-summary"] > div:nth-of-type(1) > dt + dd     v2.0    %{WAIT_LONG}
+    user waits until element contains    css:dl[data-testid="draft-version-summary"] > div:nth-of-type(2) > dt + dd     Action required    %{WAIT_LONG}
+
+    ${mapping_status}=    get text    css:dl[data-testid="draft-version-summary"] > div:nth-of-type(2) > dt + dd
+    should be equal as strings    ${mapping_status}    Action required
+
+Add headline text block to Content page
+    user navigates to content page    ${PUBLICATION_NAME}
+    user adds headlines text block
+    user adds content to headlines text block    Headline text block text
+
```

Comment:
Do we need to create this headline text block here?

Reply:
Its not mandatory required to create this headline text block, but I have added this so that I can only focus on specific errors which I am looking for. if I chose to leave this, then I would be tempted to add headline specific text block checklist errors.